feat: add asana sample (#160)
autokitteh/samples: samples/asana/README.md

---
title: Asana
description: Simple usage of the Asana API
integrations: ["asana"]
categories: ["Samples"]
---

# Asana Sample

The Asana sample demonstrates how to streamline task management using AutoKitteh's Asana integration.

It features functionality to create tasks within a workspace and update existing ones,
including changes to their names and due dates.

API details:

- [Asana API](https://developers.asana.com/docs/quick-start)
- [Python client library](https://developers.asana.com/docs/migration-guide-python-v5)

## How It Works

1. Create a new task in asana 
2. Update the name and the due date of an existing task

## Cloud Usage (Recommended)

 1. Initialize your connection through the UI

>  2. Initialize the `WORKSPACE_GID` variable,to get it log in asana and visit this URL https://app.asana.com/api/1.0/workspaces

Reviewer: ```suggestion
 2. Update the `WORKSPACE_GID` variable with the gid value from the response after logging into Asana and visiting https://app.asana.com/api/1.0/workspaces
```

 3. Edit the trigger of the workflow you want to trigger, in the "TRIGGERS" tab, under the "Actions" column
 4. Copy the provided webhook URL
 5. Send GET request to the webhook of your choice

## Trigger Workflow

- For `create_task`:
       ```shell
       curl "https://api.autokitteh.cloud/webhooks/{your-webhook-slug}" 
       ```
- For `update_task`:
       ```shell
       curl "https://api.autokitteh.cloud/webhooks/{your-webhook-slug}" \
            -d task_gid=<task_gid> \
            -d new_due_date=<new_date> \
            -d name_suffixe=<added_name>
       ```

> 

Reviewer 2: (Only) 1 empty line around headers, not 2
```suggestion
```

---
title: Asana
description: Simple usage of the Asana API
integrations: ["asana"]
categories: ["Samples"]
---

# Asana Sample

The Asana sample demonstrates how to streamline task management using AutoKitteh's Asana integration.

It features functionality to create tasks within a workspace and update existing ones,
including changes to their names and due dates.

API details:

- [Asana API](https://developers.asana.com/docs/quick-start)
- [Python client library](https://developers.asana.com/docs/migration-guide-python-v5)

## How It Works

1. Create a new task in asana 
2. Update the name and the due date of an existing task

## Cloud Usage (Recommended)

 1. Initialize your connection through the UI

>  2.  Copy the webhook trigger's URL (for step 5 below):

Reviewer 2: ```suggestion
 2. Copy the webhook trigger's URL (for the [Trigger Workflows](#trigger-workflows) section below):
```

    - Hover over the trigger's (i) icon for the webhook you want to use
    - Click the copy icon next to the webhook URL for your selected trigger
    - (Detailed instructions
      [here](https://docs.autokitteh.com/get_started/deployment#webhook-urls))

>  3. Update the `WORKSPACE_GID` variable with the gid value from the response after logging into Asana and visiting https://app.asana.com/api/1.0/workspaces

Reviewer 2: ```suggestion
 3. Update the `WORKSPACE_GID` project variable with the `gid` value from the response after logging into Asana and visiting https://app.asana.com/api/1.0/workspaces
```

>  5. Send the request to the webhook. For details see [trigger workflow](#trigger-workflow)

Reviewer 2: ```suggestion
```

> ## Trigger Workflow

Reviewer 2: ```suggestion
## Trigger Workflows
```

> - For `create_task`:

Reviewer 2: ```suggestion
`create_task`:

```

>        ```shell

Reviewer 2: Move this block to the beginning of the line, no more need for indentation

       curl -X GET "https://api.autokitteh.cloud/webhooks/{your-webhook-slug}" 
       ```

> - For `update_task`:

Reviewer 2: ```suggestion

`update_task`:

```

>        ```shell

Reviewer 2: Move this block to the beginning of the line, no more need for indentation

       curl -X POST "https://api.autokitteh.cloud/webhooks/{your-webhook-slug}" \
            -d task_gid=<task_gid> \
            -d new_due_date=<new_date> \
            -d name_suffixe=<added_name>
       ```

## Self-Hosted Deployment

Follow these [detailed instructions](https://docs.autokitteh.com/get_started/deployment) to deploy the project on a self-hosted server.

> 

Reviewer 2: ```suggestion
```
